src/krislet/Actions.java

```diff
@@ -23,21 +31,37 @@ public KickAction(String target) {
             this.target = target;
         }
 
-        public void excute(Krislet krislet) {
-            krislet.kick(100, -1); //object.m_direction); // TODO: replace -1 with object.m_direction, probs should be an argument to this action
+        public void execute(Krislet krislet) {
+        	ObjectInfo object;
+        	object = krislet.getMemory().getObject(target);
+            krislet.kick(100, object.m_direction);
         }
     }
 
     public static class MoveToAction implements Krislet.Action
     {
         private String target;
+        private static final int TARGET_DISTANCE = 10;
 
         public MoveToAction(String target) {
             this.target = target;
         }
 
-        public void excute(Krislet krislet) {
-            krislet.wait(1);
+        public void execute(Krislet krislet) {
+            ObjectInfo object;
+            object = krislet.getMemory().getObject(target);
+            while (object.m_distance > TARGET_DISTANCE)
```

Comment:
So this is having multiple commands being sent with a loop,, but there is no delay. So only the first or last (dont remember how it works) will be executed.